src/libs/ontouml2db/factory/Factory.ts

```diff
@@ -0,0 +1,316 @@
+import { GraphGeneralization } from '../graph/impl/GraphGeneralization';
+import { GraphGeneralizationSet } from '../graph/impl/GraphGeneralizationSet';
+import { Cardinality } from '../graph/util/enumerations';
+import { IGraphGeneralizationSet } from '../graph/IGraphGeneralizationSet';
+
+export class Factory {
+  private graph: IGraph;
+  private modelManager: ModelManager;
+
+  constructor(model: IPackage) {
+    let modelCopy = JSON.parse(JSON.stringify(model));
+    this.modelManager = new ModelManager(modelCopy);
+
+    this.graph = new Graph();
+  }
+
+  public mountGraph(): IGraph {
+    this.putClasses();
+
+    this.putRelations();
+
+    this.putGeneralizaitons();
+
+    this.putGeneralizationSets();
+
+    return this.graph;
+  }
+
+  /********************************************************************
+   ** puts the classes
+   *********************************************************************/
+  private putClasses(): void {
+    let classes: IClass[];
+    classes = this.modelManager.rootPackage.getAllContentsByType([
+      OntoUMLType.CLASS_TYPE,
+    ]) as IClass[];
+
+    classes.forEach((iclass: IClass) => {
+      if (this.getUfoStereotype(iclass) != null) {
+        this.putClass(iclass);
+      }
+    });
+  }
+
+  private putClass(iclass: IClass): void {
+    let node: INode;
+    let property: INodeProperty;
+
+    node = new Node(iclass.id, iclass.name, this.getUfoStereotype(iclass));
+
+    const { properties: attributes } = iclass;
+
+    if (attributes != null) {
+      for (let i = 0; i < attributes.length; i += 1) {
+        const {
+          id: attrID,
+          propertyType: attrElement,
+          name: attrName,
+          cardinality: attrCardinality,
+        } = attributes[i];
+        const { name: datatypeName } = (attrElement || {}) as IClass;
+
+        property = new NodeProperty(
+          attrID,
+          attrName,
+          datatypeName,
+          this.getAcceptNull(attrCardinality),
+          this.getIsMultivalued(attrCardinality),
+        );
+
+        node.addProperty(property);
+      }
+    }
+    this.graph.addNode(node);
+  }
+
+  private getAcceptNull(cardinality: string): boolean {
+    if (cardinality == null) return true;
+
+    if (cardinality.substring(0, 1) == '0') return true;
+    else return false;
+  }
+
+  private getIsMultivalued(cardinality: string): boolean {
+    if (cardinality == null) return false;
+
+    if (cardinality.length > 3) {
+      let num = cardinality.substring(
+        cardinality.length - 1,
+        cardinality.length,
+      );
+
+      if (num == '*') return true;
+
+      if (!isNaN(parseFloat(num))) {
+        if (Number(num) > 1) return true;
+        else return false;
+      }
+    }
+    return false;
+  }
+
+  private getUfoStereotype(iclass: IClass): ClassStereotype {
+    let classStereotype: ClassStereotype = null;
+    if (iclass.stereotypes != null) {
+      iclass.stereotypes.some(element => {
+        switch (element) {
```

Comment:
Why do you need this switch?

Reply:
Although a class can have only one UFO stereotype, the plugin allows to add more than one stereotype. Can I assume that a class has only one stereotype?